src/targets/gpu/fuse_mlir.cpp

```diff
@@ -423,6 +470,97 @@ std::vector<instruction_ref> mlir_contiguous(module_pass_manager& mpm,
     return result;
 }
 
+struct find_mlir_split_reduce
+{
+    mlir_mode conv_mode = mlir_mode::none;
+    mlir_mode dot_mode  = mlir_mode::none;
+    auto matcher() const
+    {
+        auto dot_or_conv = match::name("gpu::mlir_op");
+        // TODO: Handle reshapes inbetween
+        return mlir_split_reduce()(match::any_of[match::inputs()](dot_or_conv.bind("gemm")));
+    }
+
+    void apply(module_pass_manager& mpm, const match::matcher_result& r) const
+    {
+        auto reduce_ins = r.result;
+        auto gemm_ins   = r.instructions["gemm"];
+        auto* rm        = reduce_ins->module_inputs().front();
+        auto names      = rm->get_parameter_names();
+        std::sort(names.begin(), names.end());
+        module_ref gemm_old_mm = gemm_ins->module_inputs().front();
+        module_ref mm          = mpm.create_module(gemm_old_mm->name() + "_split_fused_reduce");
+        mm->add_instructions(gemm_old_mm);
+        mm->set_bypass();
+        std::unordered_map<instruction_ref, instruction_ref> param_map;
+        param_map[gemm_ins]      = std::prev(mm->end());
+        bool gemm_has_multi_outs = gemm_ins->outputs().size() > 1;
+        auto return_vals =
+            mm->fuse(*rm,
+                     reduce_ins->inputs(),
+                     &param_map,
+                     [&](module& main_mod,
+                         instruction_ref pos,
+                         const operation& op,
+                         const std::vector<instruction_ref>& inputs,
+                         const std::vector<module_ref>& mod_args) {
+                         if(op.name() == "pointwise")
+                         {
+                             for(const auto& skip_param : inputs)
+                             {
+                                 if(not contains(param_map, skip_param))
+                                 {
+                                     param_map[skip_param] =
+                                         skip_param; // skip adding parameter for inputs of
+                                                     // pointwise inside split_fused_reduce
+                                 }
+                             }
```

Comment:
Why is this needed? If its not in the `param_map` then that seems like a bug.

Reply:
This is fusing pointwise module arg inside the fused_mlir_module.  Note that pointwise module is submodule to `split_fused_reduce` module as well.

Inputs to pointwise would be available in `map_ins` internally to `fuse` method. https://github.com/ROCm/AMDMIGraphX/blob/ae2b02650767d53b9526f660f1b2017ec4bf56e0/src/module.cpp#L1047

But they won't be passed down to "inserter" is invoked. https://github.com/ROCm/AMDMIGraphX/blob/ae2b02650767d53b9526f660f1b2017ec4bf56e0/src/module.cpp#L260

Reply:
I see thats because you are reusing the outer param_map. I am not sure that is needed though. You could just use another param_map:

```cpp
auto param_map_2 = create_param_map_with_literals(&main_mod, sub_pm, op.compute_shape(to_shapes(inputs), mod_args));
return main_mod.fuse(*sub_pm, inputs, &param_map_2).front();
```

Reply:
I also need to add `inputs` of the pointwise op into `param_map` to avoid adding params for that. Adding to same `param_map` solves both issues.

Reply:
> Sure, you can use get_ins_param_map for that.

 No, I can't it will create map from `param -> input` it won't skip adding parameters for the inputs. 

https://github.com/ROCm/AMDMIGraphX/blob/ae2b02650767d53b9526f660f1b2017ec4bf56e0/src/module.cpp#L1004

I need map from `input --> input`

Reply:
Made some changes.